client-src/elements/chromedash-approvals-dialog.js

```diff
@@ -509,40 +497,43 @@ class ChromedashApprovalsDialog extends LitElement {
 
   handleSave() {
     const promises = [];
-    for (const fieldId of this.changedApprovalsByField.keys()) {
-      if (this.changedApprovalsByField.get(fieldId) != -1) {
-        // There shoud be only one gate by field ID for now, until we have
-        // multiple stage entities for the same type of stage, e.g.
-        // multiple shipping gates.
-        const gatesByField = this.gates.filter((g) =>
-          g.gate_type == fieldId);
-
-        if (!gatesByField.length) {
+    // Check if any votes were changed by the user, and change them if so.
+    for (const [gateId, voteValue] of this.changedApprovalsByGate) {
+      if (voteValue !== -1) {
+        const changedGate = this.gates.find(g => g.id === Number(gateId));
+        if (!changedGate) {
           continue;
         }
         promises.push(
           window.csClient.setVote(
-            this.feature.id, gatesByField[0].id,
-            this.changedApprovalsByField.get(fieldId)));
+            this.feature.id, changedGate.id,
```

Comment:
Changing the protocol between the client and server can be a problem for users who have the SPA open in a browser tab at the time that we deploy the new version.  However, this dialog was already broken so I have already told the API Owners to use the gate column.  So, I think that we can go ahead with deploying this PR in a single step.